library/network_connections.py

```diff
@@ -531,6 +531,17 @@ def ifcfg_create(
                 line = r["network"] + "/" + str(r["prefix"])
                 if r["gateway"]:
                     line += " via " + r["gateway"]
+                if connection["interface_name"]:
+                    line += " dev " + connection["interface_name"]
+                else:
+                    warn_fcn(
+                        "The connection '{0}' does not specify an interface name. "
```

Comment:
There seems to be a unit test missing for testing the route configuration without an interface_name.

Reply:
https://github.com/linux-system-roles/network/commit/ac7cc6fe94cc4fd856a323b6d0fe26e3309c2c6b shows how to check warnings for initscripts. You will have to rebase on that commit before adding a new test.

Reply:
The unit test was added. Thanks.